doc/architectural/cbmc-architecture.md

```diff
@@ -1,7 +1,139 @@
 \ingroup module_hidden
-\page cbmc-architecture CBMC Architecture
+\page cbmc-architecture CPROVER Architecture
 
-\author Martin Brain, Peter Schrammel
+\author Martin Brain, Peter Schrammel, Cesar Rodriguez
+
+This chapter offers a high level description of the different modules that
+make the CPROVER framework. At a glance, the framework comprises:
+
+* An intermediate representation (IR) called GOTO program.
+* Support to parse C, C++, and Java bytecode files into the GOTO IR.
+* Support to perform bounded model checking of GOTO programs.
+* Various static analyses, including interval analysis, alias analysis, FIXME
+
+The CPROVER framework also provides various tools that aggregate multiple
+modules together:
+
+* CBMC and JBMC: bounded model checkers for C and Java
+* `goto-analyzer`: a static analyzer and abstract interpreter of C and GOTO
+  programs.
+* `goto-cc`: a drop-in replacement for `gcc`, can be used to extract a GOTO program
```

Comment:
`goto-cc` actually has several different interface, including one for cl (Visual Studio  C/C++ compiler). Rather than 'extract', I'd say something along the lines of "compile C files to GOTO programs" maybe?